### DISCUSSION TITLE
Proposal: Deleted records should not get deleted again

### DISCUSSION THREAD

**Discussion by JanezStupar** [cachapa/sql_crdt#6] [General]
The app I am working on has a mechanism to delete some records periodically.

I noticed that deletions tend to update the previously deleted records. I think that should not be the case, because all it does really is that it inflates the changesets. 

I have already implemented this feature in `synchroflite`. Let me know if you want me to backport it to `sql_crdt`

**Comment by cachapa** [comment]
The problem with that strategy is that it may cause explicit deletions to be ignored.

CRDTs are meant for distributed local-first use cases, which means every action should be recorded so it can be properly merged.
The same reasoning applies to e.g. re-writing records even if the content is the same.

If your specific use case benefits from avoiding repeated deletions, you can implement that in your own code:

``` sql
DELETE FROM table WHERE id = ?1 AND is_deleted = 0
```

**Reply by JanezStupar** [reply]
I understand the argument.

The difference in my approach is that CRDT is hidden from the user. Thus such an explicit request could never be made in the first place. Record that was deleted does not exist in the user space of `synchroflite` at all.

Thank you for your time.
